lib/spack/spack/ci/gitlab.py

```diff
@@ -129,29 +128,50 @@ def generate_gitlab_yaml(pipeline: PipelineDag, spack_ci: SpackCIConfig, options
     # concrete environment directory, along with the spack.lock file.
     if not os.path.exists(concrete_env_dir):
         os.makedirs(concrete_env_dir)
-    shutil.copyfile(options.env.manifest_path, os.path.join(concrete_env_dir, "spack.yaml"))
-    shutil.copyfile(options.env.lock_path, os.path.join(concrete_env_dir, "spack.lock"))
 
-    update_env_scopes(
-        options.env,
-        [
-            os.path.relpath(s.path, concrete_env_dir)
-            for s in cfg.scopes().values()
-            if not s.writable
-            and isinstance(s, (cfg.DirectoryConfigScope))
-            and os.path.exists(s.path)
-        ],
-        os.path.join(concrete_env_dir, "spack.yaml"),
-        # Here transforming windows paths is only required in the special case
-        # of copy_only_pipelines, a unique scenario where the generate job and
-        # child pipelines are run on different platforms. To make this compatible
-        # w/ Windows, we cannot write Windows style path separators that will be
-        # consumed on by the Posix copy job runner.
-        #
-        # TODO (johnwparent): Refactor config + cli read/write to deal only in
-        # posix style paths
-        transform_windows_paths=(options.pipeline_type == PipelineType.COPY_ONLY),
-    )
+    # Copy the manifest and handle relative included paths
+    with open(options.env.manifest_path, "r", encoding="utf-8") as fin, open(
+        os.path.join(concrete_env_dir, "spack.yaml"), "w", encoding="utf-8"
+    ) as fout:
+        data = syaml.load(fin)
+        if "spack" not in data:
+            raise spack.config.ConfigSectionError(
+                'Missing top level "spack" section in environment'
+            )
+
+        # If there are no includes, just copy
+        if "include" in data["spack"]:
+            includes = data["spack"]["include"]
+            # If there are includes in the config, then we need to fix the relative paths
+            # to be relative from the concrete env dir used by downstream pipelines
+            env_root_path = os.path.dirname(os.path.abspath(options.env.manifest_path))
+            fixed_includes = []
+            for inc in includes:
+                if isinstance(inc, dict):
+                    inc_path = inc["path"]
+                else:
+                    inc_path = inc
+
+                if os.path.isabs(inc_path):
+                    fixed_includes.append(inc)
+                    continue
+
+                inc_abs_path = os.path.normpath(os.path.join(env_root_path, inc_path))
+                inc_path = os.path.relpath(inc_abs_path, start=concrete_env_dir)
+
+                if isinstance(inc, dict):
+                    inc["path"] = inc_path
+                else:
+                    inc = inc_path
+
+                fixed_includes.append(inc)
```

Comment:
```suggestion
                if isinstance(inc, dict):
                    inc["path"] = _rewrite_path(inc["path"], orig_root=env_root_path, new_root=concrete_env_dir)
                else:
                    inc = _rewrite_path(inc, orig_root=env_root_path, new_root=concrete_env_dir)

                fixed_includes.append(inc)
```
If we factor out the rewriting itself into a method, we can avoid needing to duplicate the same condition multiple times. Then the method for the rewrite looks like
```
def _rewrite_path(path: str, orig_root: str, new_root: str) -> str:
    if os.path.isabs(path):
        return path
    abs_path = os.path.normpath(os.path.join(orig_root, path)
    return os.path.relpath(abs_path, new_root)
```
Or if we switch to just using the absolute path
```
def _rewrite_path(path: str, orig_root: str) -> str:
    if os.path.isabs(path):
        return path
    return os.path.normpath(os.path.join(orig_root, path))
```